tiatoolbox/models/dataset/info.py

```diff
@@ -0,0 +1,129 @@
+# ***** BEGIN GPL LICENSE BLOCK *****
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software Foundation,
+# Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+#
+# The Original Code is Copyright (C) 2021, TIALab, University of Warwick
+# All rights reserved.
+# ***** END GPL LICENSE BLOCK *****
+
+
+import os
+from abc import ABC, abstractmethod
+
+from tiatoolbox import rcParam
+from tiatoolbox.utils.misc import download_data, grab_files_from_dir, unzip_data
+
+
+class ABCDatasetInfo(ABC):
+    """Define an abstract class for holding a dataset information.
+
+    Enforcing such that following attributes must always be defined by the subclass.
+
+    Attributes
+        input_list (list): A list of paths where each path points to a sample image.
+        label_list (list): A list of `int` where each is the label of the sample at
+            the same index.
+        label_name (dict): A dict indicates the possible associate name of each label
+            value.
+
+    """
+
+    @property
+    @abstractmethod
+    def input_list(self):
+        raise NotImplementedError
+
+    @property
+    @abstractmethod
+    def label_list(self):
+        raise NotImplementedError
+
+    @property
+    @abstractmethod
+    def label_name(self):
+        raise NotImplementedError
+
+
+class KatherPatchDataset(ABCDatasetInfo):
+    """Define a class for holding the Kather dataset information.
+
+    Args:
+        save_dir_path (str or None): Path to directory containing the Kather dataset,
+                 assumed to be as is after extracted. If the argument is `None`,
```

Comment:
please make sure the indentation is consistent at least in the docstring for the same function/class.